packit_service/log_versions.py

```diff
@@ -0,0 +1,46 @@
+import logging
+
+# from packit import __version__ as packit_version
+
+from ogr import __version__ as ogr_version
+from packit_service import __version__ as ps_version
+from sqlalchemy import __version__ as sqlal_version
+from flask_restx import __version__ as restx_version
+from flask import __version__ as flask_version  # type: ignore
+
+
+# For some reason mypy has an issue with flask's __version__ even though it works completely fine.
+# The "type: ignore" comment causes mypy to ignore that line.
+
+
+logger = logging.getLogger(__name__)
+
+
+# does the actual logging
+def log_package_versions(package_versions: list):
+    log_string = "\nPackage Versions:"
+    for name, version in package_versions:
+        log_string += f"\n* {name} {version}"
+    logger.info(log_string)
+
+
+# log package versions before running a job
+def log_versions_before_job():
+    package_versions = [
+        ("OGR", ogr_version),
+        ("Packit Service", ps_version),
+        ("SQL Alchemy", sqlal_version),
+    ]
+    log_package_versions(package_versions)
+
+    # NOTE: Can't log packit's version for now because it does not provide one.
```

Comment:
This comment is also out of context, it would be much more useful to put it in the `package_versions` list above.